tests/framework/extensions/charts/ranchercisbenchmark.go

```diff
@@ -0,0 +1,259 @@
+	if err != nil {
+		return err
+	}
+
+	err = wait.WatchWait(watchAppInterface, func(event watch.Event) (ready bool, err error) {
+		app := event.Object.(*catalogv1.App)
+
+		state := app.Status.Summary.State
+		if state == string(catalogv1.StatusDeployed) {
+			return true, nil
+		}
+		return false, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// Register cleanup function for uninstallation
+	client.Session.RegisterCleanupFunc(func() error {
+		return UninstallRancherCisBenchmarkChart(client, installOptions)
+	})
+
+	return nil
+}
+
+// UninstallRancherCisBenchmarkChart is a helper function that uninstalls the rancher-CIS Benchmark chart.
+func UninstallRancherCisBenchmarkChart(client *rancher.Client, installOptions *InstallOptions) error {
+	catalogClient, err := client.GetClusterCatalogClient(installOptions.ClusterID)
+	if err != nil {
+		return err
+	}
+
+	// Uninstall the Rancher CIS Benchmark chart
+	defaultChartUninstallAction := newChartUninstallAction()
+	err = catalogClient.UninstallChart(RancherCisBenchmarkName, RancherCisBenchmarkNamespace, defaultChartUninstallAction)
+	if err != nil {
+		return err
+	}
+
+	// Watch for events related to the uninstallation and wait until it is deleted
+	watchAppInterface, err := catalogClient.Apps(RancherCisBenchmarkNamespace).Watch(context.TODO(), metav1.ListOptions{
+		FieldSelector:  "metadata.name=" + RancherCisBenchmarkName,
+		TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+	})
+	if err != nil {
+		return err
+	}
+
+	err = wait.WatchWait(watchAppInterface, func(event watch.Event) (ready bool, err error) {
+		chart := event.Object.(*catalogv1.App)
+		if event.Type == watch.Error {
+			return false, fmt.Errorf("there was an error uninstalling rancher CIS Benchmark chart")
+		} else if event.Type == watch.Deleted {
+			return true, nil
+		} else if chart == nil {
+			return true, nil
+		}
+		return false, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// Uninstall the Rancher CIS Benchmark CRD chart
+	err = catalogClient.UninstallChart(RancherCisBenchmarkCRDName, RancherCisBenchmarkNamespace, defaultChartUninstallAction)
+	if err != nil {
+		return err
+	}
+
+	// Watch for events related to the uninstallation and wait until it is deleted
+	watchAppInterface, err = catalogClient.Apps(RancherCisBenchmarkNamespace).Watch(context.TODO(), metav1.ListOptions{
+		FieldSelector:  "metadata.name=" + RancherCisBenchmarkCRDName,
+		TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+	})
+	if err != nil {
+		return err
+	}
+
+	err = wait.WatchWait(watchAppInterface, func(event watch.Event) (ready bool, err error) {
+		chart := event.Object.(*catalogv1.App)
+		if event.Type == watch.Error {
+			return false, fmt.Errorf("there was an error uninstalling rancher CIS Benchmark chart")
+		} else if event.Type == watch.Deleted {
+			return true, nil
+		} else if chart == nil {
+			return true, nil
+		}
+		return false, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// Delete the Rancher CIS Benchmark namespace
```

Comment:
Can you confirm, when we delete the chart from Rancher, the namespace also gets deleted? I remember it didn't, atleast a while back.

Reply:
good point

Reply:
Yes, I tested the automation again and confirm that the namespace also gets deleted.

Reply:
Can we check/verify if, in Rancher UI, if you deploy CIS chart and then uninstall it, does it delete the namespace? Not via automation.

Reply:
From Rancher UI, the namespace does not get deleted.

Reply:
@sowmyav27 Namespace do not get deleted if we uninstall charts from rancher UI, Is that also expected in the automation?

Reply:
If we don't delete the namespace, would be able to reinstall the chart?

Reply:
We should be able to reinstall the chart as per my recollection. During installation, it will show the namespace is already existing .. I will let @vivek-shilimkar reply here tho.

Reply:
I tested, if the re-installation works, considering the namespace is already present on the cluster. It does work. Chart installs and runs the scan successfully.